Doriane Olewicki: implementation of the two challenges

olewicki opened the pull request:
Good afternoon, 
Here is my implementation of the two challenges. I am a student of Bram Adams, I know you were in contact with him. 
Thank you in advance for the attention you will give to my implementation. There is some comments at the beginning of the README.md about my code.
Best regards,
Doriane Olewicki

lbajolet commented on DiffResult.java:
```diff
@@ -0,0 +1,56 @@
+	int regions = 0;
+	//How many line were added total
+	int lineAdded = 0;
+	//How many line were deleted totla
```
Small typo here `totla`

olewicki replied:
I just noticed this typo was actually already in the DiffResult.go file. I corrected it in my file.

lbajolet commented on Main.java:
```diff
@@ -0,0 +1,216 @@
+    StartTime = System.currentTimeMillis();
+    System.out.println(computeAST());
+    EndTime = System.currentTimeMillis();
+    System.out.println(String.format("%s took %d s", "compute AST", EndTime - StartTime));
```
I feel this was meant to be factorized; why the litteral string as format argument otherwise?

olewicki replied:
I modified my code to have a factorization at that point, as well as in the rest of the code.

olewicki commented:
Regarding the number of files, I corrected it and found 1248 files. The error was in my regular expression were I forgot to grab the files without extensions.

For the indentation, I forgot to verify it was uniform over the files. I am sorry for that mistake.

The millisecond error was corrected.

For the number of calls, I counted initially the number of added calls minus the number of removed calls, to have the net increase. I am not sure what the count should be so this time I put 3 versions of count, the first one counting each call (this one is the one with which I generated the results in out.txt) and two other versions are in comments and explained in the code it-self.

Regarding the factorization, I didn't put factorization at first because I thought it was readable enough. But looking back at the code I see how wrong I was and I factorized it. Now the functions are methods of the DiffResult and AstResult classes. Some instances where added to those classes to ease the use of the methods and limit the number of parameters.

I added the command "e.printStackTrace()" to the catches, which are in fact better practice than just printing the error.
